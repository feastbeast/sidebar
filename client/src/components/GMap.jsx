import React from 'react';

const GMap = (props) => {
  let mapLink= `https://www.google.com/maps/embed/v1/place?key=AIzaSyDfVc7Wl0xQOdHnSm30Yk2lZtcBTPEZtjM&q=${props.lat},${props.lng}`;
  //https://www.google.com/maps/embed/v1/place?key=AIzaSyBFIa4aaMYhCr78dTMyKkwxvA6ABpYZW9Y&center=-33.8569,151.2152
  //https://www.google.com/maps/embed/v1/search?key=AIzaSyBFIa4aaMYhCr78dTMyKkwxvA6ABpYZW9Y&q=record+stores+in+Seattle
  return (
    
    <div className="staticMap">
      <iframe
        frameBorder="0" 
        style={{border: 0}}
        src={mapLink}
        allowFullScreen
        
      >
      </iframe>
    </div>
  )
}

export default GMap;
window.GMap = GMap;