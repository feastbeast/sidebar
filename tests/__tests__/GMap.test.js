import React from 'react';
import Enzyme from 'enzyme';
import {shallow, mount, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import GMap from '../../client/src/components/GMap';

Enzyme.configure({ adapter: new Adapter() });

describe('GMap', () => {
  it('should be defined', () => {
    let location = 'SF';
    let id = '1234';

    let wrapper = shallow(<GMap
      location={location}
      id={id}
    />) 
    expect(wrapper).toBeDefined();
  });
  it('should render correctly', () => {
    let location = 'SF';
    let id = '1234';

    let tree = shallow(<GMap
      location={location}
      id={id}
    />) 
    expect(tree).toMatchSnapshot();
  });

});