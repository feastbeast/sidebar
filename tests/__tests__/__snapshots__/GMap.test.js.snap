// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GMap should render correctly 1`] = `
<div
  className="staticMap"
>
  <iframe
    allowFullScreen={true}
    frameBorder="0"
    src="https://www.google.com/maps/embed/v1/place?key=AIzaSyDfVc7Wl0xQOdHnSm30Yk2lZtcBTPEZtjM&q=place_id:1234"
    style={
      Object {
        "border": 0,
      }
    }
  />
</div>
`;
